src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -266,118 +258,29 @@ def create_empty_extra_products(self):
-    return "compile_against_rsc({})".format(compile_target.address.spec)
-
-  def pre_compile_jobs(self, counter):
-
-    # Create a target for the jdk outlining so that it'll only be done once per run.
-    target = Target('jdk', Address('', 'jdk'), self.context.build_graph)
-    index_dir = os.path.join(self.versioned_workdir, '--jdk--', 'index')
-
-    def work_for_vts_rsc_jdk():
-      distribution = self._get_jvm_distribution()
-      jvm_lib_jars_abs = distribution.find_libs(['rt.jar', 'dt.jar', 'jce.jar', 'tools.jar'])
-      self._jvm_lib_jars_abs = jvm_lib_jars_abs
-
-      metacp_inputs = tuple(jvm_lib_jars_abs)
-
-      counter_val = str(counter()).rjust(counter.format_length(), ' ' if PY3 else b' ')
-      counter_str = '[{}/{}] '.format(counter_val, counter.size)
-      self.context.log.info(
-        counter_str,
-        'Metacp-ing ',
-        items_to_report_element(metacp_inputs, 'jar'),
-        ' in the jdk')
-
-      # NB: Metacp doesn't handle the existence of possibly stale semanticdb jars,
-      # so we explicitly clean the directory to keep it happy.
-      safe_mkdir(index_dir, clean=True)
-
-      with Timer() as timer:
-        # Step 1: Convert classpath to SemanticDB
-        # ---------------------------------------
-        rsc_index_dir = fast_relpath(index_dir, get_buildroot())
-        args = [
-          '--verbose',
-          # NB: The directory to dump the semanticdb jars generated by metacp.
-          '--out', rsc_index_dir,
-          os.pathsep.join(metacp_inputs),
-        ]
-        metacp_wu = self._runtool(
-          'scala.meta.cli.Metacp',
-          'metacp',
-          args,
-          distribution,
-          tgt=target,
-          input_files=tuple(
-            # NB: no input files because the jdk is expected to exist on the system in a known
-            #     location.
-            #     Related: https://github.com/pantsbuild/pants/issues/6416
-          ),
-          output_dir=rsc_index_dir)
-        metacp_stdout = stdout_contents(metacp_wu)
-        metacp_result = json.loads(metacp_stdout)
-
-        metai_classpath = self._collect_metai_classpath(metacp_result, jvm_lib_jars_abs)
-
-        # Step 1.5: metai Index the semanticdbs
-        # -------------------------------------
-        self._run_metai_tool(distribution, metai_classpath, rsc_index_dir, tgt=target)
-
-        self._jvm_lib_metacp_classpath = [os.path.join(get_buildroot(), x) for x in metai_classpath]
-
-      self._record_target_stats(target,
-        len(self._jvm_lib_metacp_classpath),
-        len([]),
-        timer.elapsed,
-        False,
-        'metacp'
-      )
+    return 'rsc({})'.format(compile_target.address.spec)
 
-    return [
-      Job(
-        'metacp(jdk)',
-        functools.partial(
-          work_for_vts_rsc_jdk
-        ),
-        [],
-        self._size_estimator([]),
-      ),
-    ]
+  def _zinc_key_for_target(self, compile_target):
```

Comment:
Having separately-named keys for the zinc compiles is totally the right move, I didn't mean to remove that so thanks for fixing it.